www/js/services.js

```diff
@@ -428,6 +429,160 @@ angular.module('emission.services', ['emission.plugin.logger'])
+        CAR:         0.14515,
+        BUS:         0.04751,
+        TRAIN:       0.048,
+        AIR_OR_HSR:  0.201
+      }
+    },
+    DE: {
+      regionName: "Germany",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.139,   // Tremod (passenger car)
+        BUS:         0.0535,  // Tremod (average city/coach)
+        TRAIN:       0.048,   // Tremod (average short/long distance)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    FR: {
+      regionName: "France",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.13125, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04838, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    AT: {
+      regionName: "Austria",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.14351, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04625, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    SE: {
+      regionName: "Sweden",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.13458, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04557, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    NO: {
+      regionName: "Norway",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.13265, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04185, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    },
+    CH: {
+      regionName: "Switzerland",
+      footprintData: {
+        WALKING:     0,
+        BICYCLING:   0,
+        CAR:         0.17638, // HBEFA (passenger car, considering 1 passenger)
+        BUS:         0.04866, // HBEFA (average short/long distance, considering 16/25 passengers)
+        TRAIN:       0.048,   // Tremod (DE value)
+        AIR_OR_HSR:  0.201    // Tremod (airplane)
+      }
+    }
+  };
+
+  var defaultCarbonDatasetCode = 'US';
+  var currentCarbonDatasetCode;
+
+  this.initialize = function() {
+    console.debug("Tiago: CarbonDatasetHelper.initialize()");
+    currentCarbonDatasetCode = defaultCarbonDatasetCode;
+    var value = KVStore.getDirect(CARBON_DATASET_KEY);
+    console.debug("Tiago: CarbonDatasetHelper.initialize() obtained value from storage [" + value + "]");
```

Comment:
You may want to turn some of these to Logger.log messages just in case we run into complaints. Those are logged to the app logs, which can be emailed and checked if unexpected heisenbugs occur.

Reply:
Thank you, I will.

Reply:
I'll try to keep the ones which are useful and less repetitive